### PR TITLE
feat: support zk log options & flags fix

### LINKED ISSUE #1500: sql_client: zk "exceeded deadline by N ms" floods logs
Ref https://issues.apache.org/jira/browse/ZOOKEEPER-642
We should support two feature:
1. set zk_session_timeout in sql_client(only sql_client haven't supported zk_session_timeout)
2. set log level of zk client in sql_client(servers don't need to remove the zk logs, only interactive client needs)

optional:
update zk c api version to 3.5.0, it'll reduce the amount of meaningless log messages.

### PR DESCRIPTION
**Pull request by vagetablechicken**
Closes #1500 
CLI: disable zk log by default(log level 0). use gflags zk_log_level & zk_log_file to modify `ClusterOptions`.
other sdk: use default `ClusterOptions ` now, info level and use stream stderr(default in zk).

make more flag desc.

### CODE REVIEW

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: src/zk/zk_client.h [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "boost/function.hpp"
 
 extern "C" {
+#include "stdio.h"
```
Comment (dl239) [inline src/zk/zk_client.h]: please resolve all `cpplint` warnings

File: src/zk/zk_client.h [hunk 2]
```diff
@@ -52,7 +53,11 @@ class ZkClient {
     ~ZkClient();
 
     // init zookeeper connections
-    bool Init();
+    // log level: DISABLE_LOGGING=0, ZOO_LOG_LEVEL_ERROR=1, ZOO_LOG_LEVEL_WARN=2, ZOO_LOG_LEVEL_INFO=3,
```
Comment (dl239) [inline src/zk/zk_client.h:56]: add enum

Comment (vagetablechicken) [reply src/zk/zk_client.h:56]: It's zk enum, integer is ok, no need to add more

Comment (dl239) [reply src/zk/zk_client.h:56]: it's not recommend to pass magic number when  call `init` function,

Comment (dl239) [reply src/zk/zk_client.h:56]: L44 in src/sdk/db_sdk.h and default parameter in init

**Review by dl239 — APPROVED** [review]

**Pull request merged**